### PR TITLE
Add Pattern Relations and Pattern Relation Types

### PR DESCRIPTION
**Pull request by LHommeDeBat**
#### Short Description
Adds PatternRelation and PatternRelationType with a complete CRUD-Interface to implement issue https://github.com/PlanQK/q-tal/issues/11

### CODE REVIEW

File: org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/controller/AlgorithmControllerTest.java [hunk 1]
```diff
@@ -170,6 +180,28 @@ public void initialize() {
...
+        PatternRelation patternRelation2 = new PatternRelation();
+        patternRelation2.setId(UUID.randomUUID());
+        patternRelation2.setDescription("Description2");
+        patternRelation2.setAlgorithm(algorithm1);
+        patternRelation2.setPatternRelationType(patternType2);
+        patternRelations = new HashSet<>();
+        patternRelations.add(patternRelation1);
+        patternRelations.add(patternRelation2);
+        algorithm1.setRelatedPatterns(patternRelations);
+
```
[the remark this answers is not in the document]
Comment (LHommeDeBat) [reply org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/controller/AlgorithmControllerTest.java:221]: I am assuming you mean the "initializeAlgorithms()" method. If so, then this is done.

File: org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/controller/AlgorithmControllerTest.java [hunk 2]
```diff
@@ -456,4 +488,26 @@ public void deleteAlgorithmRelation_returnOk() throws Exception {
                 + "/{relation_id}", algorithm1.getId(), algorithmRelation1.getId())).andExpect(status().isOk())
                 .andReturn();
     }
+
+    @Test
+    public void getPatternRelations_returnTwo() throws Exception {
+        when(algorithmService.findById(any())).thenReturn(algorithm1);
```
[the remark this answers is not in the document]
Comment (LHommeDeBat) [reply org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/controller/AlgorithmControllerTest.java:517]: done

File: org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/controller/AlgorithmControllerTest.java [hunk 3]
```diff
@@ -456,4 +488,26 @@ public void deleteAlgorithmRelation_returnOk() throws Exception {
...
+                .andExpect(status().isOk()).andReturn();
+
+        var resultList = ObjectMapperUtils.mapResponseToList(result.getResponse().getContentAsString(),
+                "patternRelationDtoes", PatternRelationDto.class);
+        assertEquals(2, resultList.size());
+    }
+
+    @Test
+    public void getPatternRelations_returnNotFound() throws Exception {
+        when(algorithmService.findById(any())).thenThrow(NoSuchElementException.class);
```
[the remark this answers is not in the document]
Comment (LHommeDeBat) [reply org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/controller/AlgorithmControllerTest.java:531]: done

**Comment by LHommeDeBat** [comment]
@c-mueller I think I adjusted everything you mentioned.

File: org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/controller/AlgorithmControllerTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (c-mueller) [reply org.planqk.atlas.web/src/test/java/org/planqk/atlas/web/controller/AlgorithmControllerTest.java:221]: Yes exactly!

**Comment by LHommeDeBat** [comment]
> 
> 
> Looks good to me!

OK, I'm merging it then

**Pull request merged**